src/main/java/org/myrobotlab/service/interfaces/NeoPixelControl.java

```diff
@@ -30,69 +30,45 @@
 public interface NeoPixelControl extends NameProvider {
 
   /**
-   * high level "attach" which internally will call attachDevice(Device device,
-   * int[] config)
-   * 
+   * explicit attach no additional parameters
    * @param controller
-   *          c
-   * 
-   * @param numPixel
-   *          - All of the config needed for the device -numPixel=number of
-   *          pixel of the neopixel hardware
-   * @param pin
-   *          p -
-   * @throws Exception
-   *           e
    */
-  public void attach(NeoPixelController controller, int pin, int numPixel) throws Exception;
+  public void attachNeoPixelController(NeoPixelController controller);
 
-  /*
-   * high level "detach" with internally will call detachDevice(Device device)
+  /**
+   * explicit detach
+   * @param controller
    */
-  public void detach(NeoPixelController controller);
+  public void detachNeoPixelController(NeoPixelController controller);
 
   /**
    * SetPixel: defining the pixels of the pixel matrix. Setting pixels are not
    * send directly to the Neopixel Hardware but send with the sendPixelMatrix()
    * method
    * 
    * @param address
-   *          - value 1 to numPixel
    * @param red
-   *          - value 0-255
    * @param green
-   *          - value 0-255
    * @param blue
-   *          - value 0-255
    */
   public void setPixel(int address, int red, int green, int blue);
 
   /**
-   * Send a matrix of pixel to the neopixel hardware
+   * Send a matrix of pixels to the neopixel hardware

```

Comment:
this method should take an array of pixels if that's what it does.

Reply:
its to set a single pixel - that's why its named setPixel